runtime/sys.c

```diff
@@ -416,17 +423,16 @@ CAMLprim value caml_sys_getenv(value var)
   return val;
 }
 
-char_os * caml_exe_name;
 static value main_argv;
 
 CAMLprim value caml_sys_get_argv(value unit)
 {
   CAMLparam0 ();   /* unit is unused */
   CAMLlocal2 (exe_name, res);
-  exe_name = caml_copy_string_of_os(caml_exe_name);
+  exe_name = caml_copy_string_of_os(caml_params->exe_name);
   res = caml_alloc_small(2, 0);
-  Field(res, 0) = exe_name;
-  Field(res, 1) = main_argv;
+  caml_initialize(&Field(res, 0), exe_name);
```

Comment:
Can you `Field` as l-value for assignment, as in trunk.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/9ef07afa0adc5b5d764c2a1e59c9551bb6a0e5f6